src/main/java/it/eng/idsa/dataapp/web/rest/exceptions/DataAppExceptionHandler.java

```diff
@@ -0,0 +1,186 @@
+			throws IOException {
+
+		logger.info("handleNotFoundException");
+		logger.error("Message: " + ex.getMessage());
+
+		Message errorMessage = createErrorMessage(ex.getHeader(), RejectionReason.NOT_FOUND);
+
+		return buildErrorResponse(errorMessage);
+	}
+
+	@ExceptionHandler(MalformedMessageException.class)
+	protected ResponseEntity<Object> handleMalformedMessageException(MalformedMessageException ex,
+			HttpServletRequest req) throws IOException {
+
+		logger.info("handleMalformedMessageException");
+		logger.error("Message: " + ex.getMessage());
+
+		Message errorMessage = createErrorMessage(ex.getHeader(), RejectionReason.MALFORMED_MESSAGE);
+
+		return buildErrorResponse(errorMessage);
+	}
+
+	@ExceptionHandler(NotAuthenticatedException.class)
+	protected ResponseEntity<Object> handleNotAuthenticatedException(NotAuthenticatedException ex,
+			HttpServletRequest req) throws IOException {
+
+		logger.info("handleNotAuthenticatedException");
+		logger.error("Message: " + ex.getMessage());
+
+		Message errorMessage = createErrorMessage(ex.getHeader(), RejectionReason.NOT_AUTHENTICATED);
+
+		return buildErrorResponse(errorMessage);
+	}
+
+	@SuppressWarnings("unchecked")
+	private ResponseEntity<Object> buildErrorResponse(Message header) throws IOException {
+		if (StringUtils.equals("http-header", httpConfig)) {
+			HttpHeaders responseHeaders = new HttpHeaders();
+			Map<String, Object> responseHeaderMap = new HashMap<>();
+			responseHeaderMap = HttpHeadersUtil.messageToHttpHeaders(header);
+			responseHeaders = HttpHeadersUtil.createResponseMessageHttpHeaders(responseHeaderMap);
+
+			ResponseEntity<?> response = ResponseEntity.noContent().headers(responseHeaders).build();
+
+			return (ResponseEntity<Object>) response;
+		} else {
+			ContentType payloadContentType = ContentType.TEXT_PLAIN;
+
+			// prepare body response - multipart message.
+			HttpEntity resultEntity = messageUtil.createMultipartMessageForm(
+					MultipartMessageProcessor.serializeToJsonLD(header), null, payloadContentType);
+
+			String contentType = resultEntity.getContentType().getValue();
+			contentType = contentType.replace("multipart/form-data", "multipart/mixed");
+
+			ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+			resultEntity.writeTo(outStream);
+			outStream.flush();
+
+			return ResponseEntity.ok().header("foo", "bar")
+					.contentType(MediaType.parseMediaType(resultEntity.getContentType().getValue()))
+					.body(outStream.toString());
+		}
+	}
+
+	private Message createErrorMessage(Message header, RejectionReason rejectionReason) {
+
+		if (StringUtils.equals("http-header", httpConfig)) {
+
+			HttpHeaders headers = new HttpHeaders();
+			String responseMessageType = "ids:RejectionMessage";
+			DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+			Date date = new Date();
+			String formattedDate = dateFormat.format(date);
+
+			headers.add("ids-messagetype", responseMessageType);
```

Comment:
Do we need all those headers.add, or this will be handled by HttpHeaderUtil.messageToHeaders method?